akka-projection-cassandra/src/main/scala/akka/projection/cassandra/internal/CassandraProjectionImpl.scala

```diff
@@ -150,20 +148,18 @@ import akka.stream.scaladsl.Source
    * This internal class will hold the KillSwitch that is needed
    * when building the mappedSource and when running the projection (to stop)
    */
-  private class InternalProjectionState(settings: ProjectionSettings)(
-      implicit systemProvider: ClassicActorSystemProvider) {
+  private class InternalProjectionState(settings: ProjectionSettings)(implicit system: ActorSystem[_]) {
 
     private val killSwitch = KillSwitches.shared(projectionId.id)
 
     private[projection] def mappedSource(): Source[Done, _] = {
-      val system: ActorSystem = systemProvider.classicSystem
       // FIXME maybe use the session-dispatcher config
-      implicit val ec: ExecutionContext = system.dispatcher
+      implicit val ec: ExecutionContext = system.classicSystem.dispatcher
 
-      val logger = Logging(systemProvider.classicSystem, this.getClass)
+      val logger = Logging(system.classicSystem, this.getClass)
 
       // FIXME session lookup could be moved to CassandraOffsetStore if that's better
-      val session = CassandraSessionRegistry(system).sessionFor(sessionConfigPath)
+      val session = CassandraSessionRegistry(system.classicSystem).sessionFor(sessionConfigPath)
```

Comment:
are you sure that it doesn't work with `CassandraSessionRegistry(system)`? I thought we added support for that in extensions?

Reply:
I think so, I will recheck